### PR TITLE
Add APP_PROTOCOL env var for app

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
This PR is the companion of https://github.com/dapr/dapr/pull/6512 which makes the same change in the Dapr Injector.

It adds the `APP_PROTOCOL` env var for the app that is being started with `dapr run`. As explained in the PR in dapr/dapr, this is helpful for the app to know if it should start a server with HTTPS or H2C enabled.

### CODE REVIEW

File: pkg/standalone/run.go
Comment (ItalyPaleAle) [reply pkg/standalone/run.go:408]: We don't normally error in the injector, because that causes the injection to fail and it's hard to debug. I'd let daprd handle the invalid value on startup

**Comment by ItalyPaleAle** [comment]
Can this be re-opened please? Also, is there something blocking this PR from being merged? @dapr/maintainers-cli @dapr/approvers-cli

**Review by mukundansundar — APPROVED** [review]

**Pull request merged**
